### PR TITLE
Oxidize api version

### LINKED ISSUE #3742: Oxidize `ApiVersion`

### PR DESCRIPTION
**Pull request by Volham22**
# What has changed ?

 This closes #3742 by adding python bindings for `ApiVersion` type.

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: parsec/serde/fields.py [hunk 1]
```diff
@@ -264,6 +265,17 @@ def _deserialize(self, value: object, attr: str, data: dict[str, object]) -> RsD
         return value
 
 
+class ApiVersion(Field[RsApiVersion]):
+    """ApiVersion already handled by pack/unpack"""
+
+    def _deserialize(self, value: object, attr: str, data: dict[str, object]) -> RsApiVersion:
+        if not isinstance(value, RsApiVersion):
+            breakpoint()
```
Comment (TimeEngineer) [inline parsec/serde/fields.py:273]: Remove

File: oxidation/libparsec/crates/protocol/src/handshake.rs [hunk 2]
```diff
@@ -66,6 +76,26 @@ impl Display for ApiVersion {
     }
 }
 
+impl TryFrom<&str> for ApiVersion {
+    type Error = &'static str;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let (version_str, revision_str) = value
+            .split_once('.')
+            .ok_or("Api version string must be follow this pattern `<version>.<revision>`")?;
```
Comment (TimeEngineer) [inline oxidation/libparsec/crates/protocol/src/handshake.rs:85]: You aren't doing the same thing here, you don't check if there are more dots...

**Comment by TimeEngineer** [comment]
Fix `mypy` for QA

**Comment by Volham22** [comment]
> Fix `mypy` for QA

I tried. This is annoying because running `mypy parsec` and then `pre-commit run -a` fix the issue. :thinking:

**Comment by Volham22** [comment]
We should wait for #3754 to be merged and rebase. QA should be fixed then

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**